### ISSUE TITLE
False positive on Style/HashEachMethods

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12509]
## Expected behavior

When the block uses array deconstruction, the cop needs to identify variables that are used from that deconstruction.

## Actual behavior

It doesn't identify the variables, and autocorrect will completely remove the argument, causing a runtime NameError.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

last_comment_dates.each do |(user_id, _author_id), date|
  next unless (student = data[user_id])

  student[:last_interaction] = [student[:last_interaction], date].compact.max
end
```

`rubocop --debug -- test.rb`:
```
For /Users/cody/src: Default configuration from /Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.58.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/cody/src/test.rb
Loading cache from /Users/cody/.cache/rubocop_cache/eaf5d3861d1bca58d12ea7599b7aabb0f6df4dad/6d7a3b621ca1730e04accd938619e4bdab66cfb1/57bb0ed8d7f0a9cef6ba21aa748af0d36bc76818
C

Offenses:

test.rb:3:1: C: [Corrected] Style/HashEachMethods: Use each_value instead of each and remove the unused (user_id, _author_id) block argument.
last_comment_dates.each do |(user_id, _author_id), date| ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
Finished in 0.1010999999998603 seconds
```

Resulting file:

```ruby
# frozen_string_literal: true

last_comment_dates.each_value do |date|
  next unless (student = data[user_id])

  student[:last_interaction] = [student[:last_interaction], date].compact.max
end
```

Note that `user_id` is no longer defined.

## RuboCop version

```
% rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin21]
```

**Comment by koic** [comment]
This issue is a dup with #12441 and resolved by #12443.
